controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -576,7 +595,29 @@ func validateControlPlaneSpec(rosaClient *rosa.RosaClient, rosaScope *scope.ROSA
 		return false, nil
 	}
 
+	// validate Etcd Encryption KMS Arn
+	err = kmsArnRegexpValidator.ValidateKMSKeyARN(&rosaScope.ControlPlane.Spec.EtcdEncryptionKMSArn)
```

Comment:
move this to line 369 after the if check

Reply:
This follows the current implementation routine. There is separate function validateControlPlaneSpec for handling all rosa control plane spec validation. The validateControlPlaneSpec function is triggered before creating the cluster

Reply:
No need to validate the arn kms, if it is not set.

Reply:
This is implemented inside of the function ValidateKMSKeyARN - if kmsKeyARN is not set, it return nil directly

https://github.com/openshift-online/ocm-common/blob/ff8c6c11d90947a20612d87b2b6435c72fe85db4/pkg/resource/validations/kms_arn_regex_validation.go#L13-L14